File: Explorer/TourService/Models/Tour.cs
namespace TourService.Models;

public class Tour
{
    public int Id { get; set; }
    public int AuthorId { get; set; }
    public required string AuthorUsername { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required string Difficulty { get; set; }
    public string[] Tags { get; set; } = Array.Empty<string>();
    public TourStatus Status { get; set; }
    public decimal Price { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public ICollection<KeyPoint> KeyPoints { get; set; } = new List<KeyPoint>();
}

File: Explorer/ExplorerFrontend/src/shared/types/tour.ts
export type KeyPointResponse = {
  id: number
  tourId: number
  name: string
  description: string
  latitude: number
  longitude: number
  imageUrl?: string | null
  orderIndex: number
}

export type TransportType = 'Walking' | 'Bicycle' | 'Car'

export type TourTravelTimeResponse = {
  transportType: TransportType
  durationMinutes: number
}

export type RoutePointResponse = {
  latitude: number
  longitude: number
}

export type TourResponse = {
  id: number
  authorId: number
  authorUsername: string
  name: string
  description: string
  difficulty: string
  tags: string[]
  status: string
  price: number
  lengthKm: number
  createdAtUtc: string
  publishedAtUtc?: string | null
  archivedAtUtc?: string | null
  keyPoints: KeyPointResponse[]
  travelTimes: TourTravelTimeResponse[]
}
